// @generated/type-graphql/resolvers/relations/Workout/WorkoutRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkoutRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
const Exercise_1 = require("../../../models/Exercise");
const User_1 = require("../../../models/User");
const Workout_1 = require("../../../models/Workout");
const WorkoutExercisesArgs_1 = require("./args/WorkoutExercisesArgs");
const helpers_1 = require("../../../helpers");
let WorkoutRelationsResolver = class WorkoutRelationsResolver {
    async exercises(workout, ctx, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.findUnique({
            where: {
                id: workout.id,
            },
        }).exercises(args);
    }
    async createdBy(workout, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.findUnique({
            where: {
                id: workout.id,
            },
        }).createdBy({});
    }
};
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => [Exercise_1.Exercise], {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Workout_1.Workout, Object, WorkoutExercisesArgs_1.WorkoutExercisesArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutRelationsResolver.prototype, "exercises", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => User_1.User, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Workout_1.Workout, Object]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutRelationsResolver.prototype, "createdBy", null);
WorkoutRelationsResolver = (0, tslib_1.__decorate)([
    TypeGraphQL.Resolver(_of => Workout_1.Workout)
], WorkoutRelationsResolver);
exports.WorkoutRelationsResolver = WorkoutRelationsResolver;
