// @generated/type-graphql/enums/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkoutScalarFieldEnum = exports.UserScalarFieldEnum = exports.SortOrder = exports.QueryMode = exports.ExerciseScalarFieldEnum = exports.Category = exports.BodyPart = void 0;
var BodyPart_1 = require("./BodyPart");
Object.defineProperty(exports, "BodyPart", { enumerable: true, get: function () { return BodyPart_1.BodyPart; } });
var Category_1 = require("./Category");
Object.defineProperty(exports, "Category", { enumerable: true, get: function () { return Category_1.Category; } });
var ExerciseScalarFieldEnum_1 = require("./ExerciseScalarFieldEnum");
Object.defineProperty(exports, "ExerciseScalarFieldEnum", { enumerable: true, get: function () { return ExerciseScalarFieldEnum_1.ExerciseScalarFieldEnum; } });
var QueryMode_1 = require("./QueryMode");
Object.defineProperty(exports, "QueryMode", { enumerable: true, get: function () { return QueryMode_1.QueryMode; } });
var SortOrder_1 = require("./SortOrder");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return SortOrder_1.SortOrder; } });
var UserScalarFieldEnum_1 = require("./UserScalarFieldEnum");
Object.defineProperty(exports, "UserScalarFieldEnum", { enumerable: true, get: function () { return UserScalarFieldEnum_1.UserScalarFieldEnum; } });
var WorkoutScalarFieldEnum_1 = require("./WorkoutScalarFieldEnum");
Object.defineProperty(exports, "WorkoutScalarFieldEnum", { enumerable: true, get: function () { return WorkoutScalarFieldEnum_1.WorkoutScalarFieldEnum; } });
