// @generated/type-graphql/resolvers/crud/Workout/WorkoutCrudResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkoutCrudResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
const graphql_fields_1 = (0, tslib_1.__importDefault)(require("graphql-fields"));
const AggregateWorkoutArgs_1 = require("./args/AggregateWorkoutArgs");
const CreateManyWorkoutArgs_1 = require("./args/CreateManyWorkoutArgs");
const CreateWorkoutArgs_1 = require("./args/CreateWorkoutArgs");
const DeleteManyWorkoutArgs_1 = require("./args/DeleteManyWorkoutArgs");
const DeleteWorkoutArgs_1 = require("./args/DeleteWorkoutArgs");
const FindFirstWorkoutArgs_1 = require("./args/FindFirstWorkoutArgs");
const FindManyWorkoutArgs_1 = require("./args/FindManyWorkoutArgs");
const FindUniqueWorkoutArgs_1 = require("./args/FindUniqueWorkoutArgs");
const GroupByWorkoutArgs_1 = require("./args/GroupByWorkoutArgs");
const UpdateManyWorkoutArgs_1 = require("./args/UpdateManyWorkoutArgs");
const UpdateWorkoutArgs_1 = require("./args/UpdateWorkoutArgs");
const UpsertWorkoutArgs_1 = require("./args/UpsertWorkoutArgs");
const helpers_1 = require("../../../helpers");
const Workout_1 = require("../../../models/Workout");
const AffectedRowsOutput_1 = require("../../outputs/AffectedRowsOutput");
const AggregateWorkout_1 = require("../../outputs/AggregateWorkout");
const WorkoutGroupBy_1 = require("../../outputs/WorkoutGroupBy");
let WorkoutCrudResolver = class WorkoutCrudResolver {
    async workout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.findUnique({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async findFirstWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.findFirst({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async workouts(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.findMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async createWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.create({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async createManyWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.createMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async deleteWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.delete({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async updateWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.update({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async deleteManyWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.deleteMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async updateManyWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.updateMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async upsertWorkout(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.upsert({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async aggregateWorkout(ctx, info, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.aggregate({
            ...args,
            ...(0, helpers_1.transformFields)((0, graphql_fields_1.default)(info)),
        });
    }
    async groupByWorkout(ctx, info, args) {
        const { _count, _avg, _sum, _min, _max } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).workout.groupBy({
            ...args,
            ...Object.fromEntries(Object.entries({ _count, _avg, _sum, _min, _max }).filter(([_, v]) => v != null)),
        });
    }
};
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => Workout_1.Workout, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, FindUniqueWorkoutArgs_1.FindUniqueWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "workout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => Workout_1.Workout, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, FindFirstWorkoutArgs_1.FindFirstWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "findFirstWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => [Workout_1.Workout], {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, FindManyWorkoutArgs_1.FindManyWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "workouts", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Workout_1.Workout, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, CreateWorkoutArgs_1.CreateWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "createWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, CreateManyWorkoutArgs_1.CreateManyWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "createManyWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Workout_1.Workout, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, DeleteWorkoutArgs_1.DeleteWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "deleteWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Workout_1.Workout, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, UpdateWorkoutArgs_1.UpdateWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "updateWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, DeleteManyWorkoutArgs_1.DeleteManyWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "deleteManyWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, UpdateManyWorkoutArgs_1.UpdateManyWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "updateManyWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Workout_1.Workout, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, UpsertWorkoutArgs_1.UpsertWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "upsertWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => AggregateWorkout_1.AggregateWorkout, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, AggregateWorkoutArgs_1.AggregateWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "aggregateWorkout", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => [WorkoutGroupBy_1.WorkoutGroupBy], {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, GroupByWorkoutArgs_1.GroupByWorkoutArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], WorkoutCrudResolver.prototype, "groupByWorkout", null);
WorkoutCrudResolver = (0, tslib_1.__decorate)([
    TypeGraphQL.Resolver(_of => Workout_1.Workout)
], WorkoutCrudResolver);
exports.WorkoutCrudResolver = WorkoutCrudResolver;
